sampleproject/components/greeting.py

```diff
@@ -1,13 +1,29 @@
+from typing import Any, Dict
+
+from django.http import HttpResponse
+
 from django_components import component
 
 
 @component.register("greeting")
-class greeting(component.Component):
-    def get_context_data(self, greet, *args, **kwargs):
-        return {"greet": greet}
+class Greeting(component.Component):
+    def get(self, request, *args, **kwargs):
+        context = self.get_context_data(request, *args, **kwargs)
+        slots = [("message", "Hello, world!", None)]
+        self.fill_slots(slots)
+        rendered_component = self.render(context)
+        return HttpResponse(rendered_component)
+
+    def get_context_data(self, request=None, **kwargs) -> Dict[str, Any]:
+        if request:
+            name = request.GET.get("name", "") if request else ""
+        else:
+            name = kwargs.get("name", "")
+        return {"name": name}
```

Comment:
I think I'd like to keep the get_context_data method free of the request object. I think it makes more sense to handle request related logic in the get method instead.

Reply:
Yes, definitely. There's no real reason for it to be here!